Add flag to download_checkmd5 to ignore network errors

trainman419 opened the pull request:
Convert catkin_checkmd5 to argparse and add a flag to make it exit normally if the network connection isn't available.

trainman419 committed:
Convert argument parsing to argparse

trainman419 committed:
Add flag to exit zero on network errors

wjwwood commented:
+1

wjwwood commented on cmake/test/download_checkmd5.py:
```diff
@@ -127,24 +127,30 @@ def main(argv=sys.argv[1:]):
+    parser.add_argument("uri")
+    parser.add_argument("dest")
+    parser.add_argument("md5sum", nargs='?')
+    parser.add_argument("--non-essential", action="store_true")
```
Should there be a `default=False` parameter here? Maybe the default `default` will work, I'm not sure.

trainman419 replied:
The default here is automatically false: https://docs.python.org/2/library/argparse.html#action

wjwwood replied:
Neat, I always thought the default was _always_ `None`.

dirk-thomas replied:
Please add a help text for this argument.

May be additionally a different name would also make the functionality more obvious - may be: `--ignore-error`

dirk-thomas commented on cmake/test/download_checkmd5.py:
```diff
@@ -127,24 +127,30 @@ def main(argv=sys.argv[1:]):
+    uri = args.uri
+    dest = args.dest
+    md5sum = args.md5sum
+    essential = not args.non_essential
```
Please avoid these four lines und just use `args.xxx` later in the code.

dirk-thomas commented on cmake/test/download_checkmd5.py:
```diff
@@ -127,24 +127,30 @@ def main(argv=sys.argv[1:]):
+            if essential:
+                raise
+            else:
+                sys.exit(0)
```
Please avoid the extra `else`. Since the boolean logic is inverted (`non_essential` / `essential`) I would also flip the if / else parts like this:

```
if args.non_essential:
    sys.exit(0)
raise
```

The same several lines below.

dirk-thomas commented:
Is this pull request still valid and will be updated based on the existing comments or should I close it since it is not required anymore?

trainman419 commented:
I've been busy. I'll update this to follow the proper coding conventions.

trainman419 commented:
I've addressed most of the comments.

I've opted to keep the `uri` variable because it gets modified.

dirk-thomas commented:
Since the main functions has a lot of different return codes now can you please make one last change. When invoking `main()` please wrap this into `sys.exit()`:

```
sys.exit(main())
```

And in the main function convert all `sys.exit` calls to simple returns.

Please also squash your changes afterwards before merging. Thanks!

trainman419 commented:
k

dirk-thomas commented:
Please look into the Travis failure.

trainman419 commented:
It looks like the test is failing because it expects main() to call sys.exit()

I guess I'll update the test

On December 15, 2014 12:43:02 PM Dirk Thomas notifications@github.com wrote:

> Please look into the Travis failure.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/ros/catkin/pull/637#issuecomment-67061965

trainman419 committed:
Fix test case

trainman419 commented:
Test updated. I'm going to wait for the tests to pass before I squash this.

trainman419 commented:
I thought this way was more pythonic. See https://github.com/ros-visualization/rqt_common_plugins/pull/292

wjwwood commented:
I would suggest `assertNotEqual(main(...), 0)` since you have an expected value or even `assertEqual(main(...), <some ret value you expect>)`.

I believe it is more pythonic to evaluate a list or tuple without `len(...)`. When evaulating the logical value of a list something specific is expected, but in this case is `main(...)` supposed to return a boolean, an integer, a string (empty or not), a list (empty or not), None, all of these, some subset? It's not clear what you're testing for. That being said, either is probably fine, and pretty easy to understand with some context.

trainman419 committed:
Fix style in download_checkmd5 test.

Make main always return 0 on success.
Test that the md5 sum check returns nonzero on error

trainman419 commented:
Done. Again.

I've also updated main so that it always returns 0 on success.

wjwwood commented:
+1

dirk-thomas commented:
Thank you. Squashed and merged in: 8033824d340d801eb7a9b2a94504b501173aaff9
